ddtrace/appsec/_remoteconfiguration.py

```diff
@@ -36,6 +39,10 @@ def __init__(self, _preprocess_results, callback):
         self._subscriber = self.__subscriber_class__(self.__shared_data__, callback, "ASM")
 
 
+def _forksafe_appsec_rc():
+    remoteconfig_poller.start_subscribers_by_product(APPSEC_PRODUCTS)
```

Comment:
Looks good. I'm just wondering if we can do something similar to what the debugger does, that is, first disable appsec, and then re-enable by simply re-registering the product. That seems to do the trick for DI, without the need for this extra method.

Reply:
`start_subscribers_by_product` disables and re-enables it. `start_subscribers_by_product` starts all products that we registered previously. If the product is not registered, `start_subscribers_by_product` does nothing. We can't do what you suggest because ASM enables and disables products dynamically. `ASM_DATA`, `ASM`, and `ASM_DD` depend on the callbacks that we receive from the `ASM_FEATURE` product callbacks.